cpp/src/arrow/engine/substrait/util.h

```diff
@@ -25,6 +25,8 @@
 #include "arrow/engine/substrait/options.h"
 #include "arrow/util/iterator.h"
 
+#include "substrait/algebra.pb.h"  // IWYU pragma: export
```

Comment:
This part is not very sure. We could include proto headers which is required for this, but what is the best practice here? Since there would be more utils like this which could reference the interfaces in the `algebra.pb.h`, we could use it. But as per this PR what is the best?

Reply:
Unfortunately, we can't do this.  I made the same mistake playing around with extension rels.  However, this is part of the public API (these methods are used by python).

I agree it would be useful to have this method in a common spot.  Can you make a `util_internal.h`  for this?

Reply:
@westonpace I added and internal file, could you please check?